rutil/Errdes.hxx

```diff
@@ -0,0 +1,331 @@
+/* 
+This header file contains functions that will return error message associated with error number/code from following functions :
+
+-> SSL_ERROR_SYSCALL  [http://www.cl.cam.ac.uk/cgi-bin/manpage?3+errno]    
+-> SSL_get_error [https://www.openssl.org/docs/manmaster/ssl/SSL_get_error.html]
+-> X509_verify_cert_error_string(verifyErrorCode) [https://www.openssl.org/docs/manmaster/crypto/X509_STORE_CTX_get_error.html]
+
+*/
+
+/*  Q. Why strerror(int errno) function is not used ? (http://www.cplusplus.com/reference/cstring/strerror/)
+    A. If error code is 22, strerror(int errno) will print "Invalid argument". But this is not so convenient, instead of using this function,
+       I have created function 'errortostringXX(int errno)' which will print error description in the format "EINVAL (Invalid argument)" 
+       which is more easy to read and understand.
+ */
+
+/*----------------END-------------------END-------------------END-------------------END---------------------END-------------------END----------*/
+
+#pragma GCC diagnostic ignored "-Wwrite-strings"
+
+#ifndef ERRDES_HXX
+#define ERRDES_HXX
+
+#include "Errdes.hxx"
+#include <cstdio>
+#include <cstring>
+
+#define MAX_ERROR_LEN 256
+static char buf[MAX_ERROR_LEN]; 
+
+/*           Below are the error description from SSL_ERROR_SYSCALL or strerror(int errno) [http://www.cl.cam.ac.uk/cgi-bin/manpage?3+errno]            */
+
+static char* messagesOS[] = {
+/*0               0   */  "No error 0",
+/*EPERM           1   */  "EPERM (Operation not permitted) 1",
```

Comment:
these numeric values are not safe to be hard-coded or used as indexes into the array